src/tensor_products.jl

```diff
@@ -345,9 +330,23 @@ julia> tdot(A)
  0.48726  0.540229  0.190334
 ```
 """
-@inline function tdot{dim}(S1::Tensor{2, dim})
-    return SymmetricTensor{2, dim}(transpdot(get_data(S1)))
+@generated function tdot{dim}(S1::Tensor{2, dim})
```

Comment:
moved this inline, instead of having `transpdot` and `dot_matmatT`  in util files.